libraries/stdlib/stdlib_ng.mtlx

```diff
@@ -1160,6 +1160,328 @@
     <output name="out" type="float" nodename="N_range" />
   </nodegraph>
 
+  <!--
+    Node: <randomfloat>
+    Produces a randomized float, based on an 'input' signal and 'seed' value.
+  -->
+  <nodegraph name="NG_randomfloat_float" nodedef="ND_randomfloat_float">
+    <multiply name="N_offset_seed" type="float">
+      <input name="in1" type="float" interfacename="seed" />
+      <input name="in2" type="float" value="43758.5" />
+    </multiply>
+    <subtract name="N_subtract1" type="float">
+      <input name="in1" type="float" interfacename="in" />
+      <input name="in2" type="float" nodename="N_offset_seed" />
+    </subtract>
+    <combine2 name="N_combine2" type="vector2">
+      <input name="in1" type="float" nodename="N_subtract1" />
+      <input name="in2" type="float" nodename="N_offset_seed" />
+    </combine2>
+    <cellnoise2d name="N_cellnoise_1" type="float">
+      <input name="texcoord" type="vector2" nodename="N_combine2" />
+    </cellnoise2d>
+    <range name="N_remapRange" type="float">
+      <input name="in" type="float" nodename="N_cellnoise_1" />
+      <input name="outlow" type="float" interfacename="min" />
+      <input name="outhigh" type="float" interfacename="max" />
+      <input name="doclamp" type="boolean" value="true" />
+    </range>
+    <output name="out" type="float" nodename="N_cellnoise_1" />
```

Comment:
Shouldn't this node return the remapped value computed by `N_remapRange`?

Reply:
Thanks @jstone-lucasfilm you're right. That's a mistake, will fix that momentarily.

BTW, I noticed you reverted the in min/max back to their defaults. Which makes sense in many cases. But in this case, where I expect the input to be 0-1, doesn't relying on the default values make the node more fragile? If remap were to change the default min/max, for example to -1 to 1, any nodes assuming the old default would break...

Reply:
Thanks Chris!

For the default values, we don't allow them to change over time, so it should safe to omit default-valued inputs.  In the rare case when node signature changes are required in a new version of MaterialX, we add explicit upgrade logic that maintains the legacy behavior of older assets.  (This is one reason that we take such pains to get the interfaces right, since they can not be changed without writing upgrade code that maintains the original behavior for older assets.)